internal/pkg/agent/application/paths/paths_darwin.go

```diff
@@ -10,6 +10,10 @@ const (
 	// BinaryName is the name of the installed binary.
 	BinaryName = "elastic-agent"
 
+	// DevelopmentBinaryName is the name of the installed binary when --develop
+	// flag is used
+	DevelopmentBinaryName = "elastic-development-agent"
+
```

Comment:
I don't think you can do this, as the `--develop` is just an alias for the `--namespace` flag. That allows multiple agents to be installed on the same host at the same time. The name of the executable here should not be hard coded as it should either come from the `--namespace` flag or from infering the namespace by reading the namespace from the path that is loaded using the full filepath from `os.Args[0]`.